game_patch/input/mouse.cpp

```diff
@@ -115,30 +115,36 @@ ConsoleCommand2 ms_cmd{
 float scope_sensitivity_factor = 1.0f;
 float scanner_sensitivity_factor = 1.0f;
 
+constexpr float clamp_sensitivity_modifier(float modifier)
+{
+    constexpr const float min = 0.01f; // prevent division by zero in scope sens
+    constexpr const float max = 4.0f;
+    return std::clamp(modifier, min, max);
+}
+
 void patch_scope_and_scanner_sensitivity()
 {
-    AsmWriter{0x004309B1}.fmul<float>(AsmRegMem(&scope_sensitivity_factor));
-    AsmWriter{0x004309DE}.fmul<float>(AsmRegMem(&scanner_sensitivity_factor));
+    AsmWriter{0x004309B1}.fmul<float>(AsmRegMem{&scope_sensitivity_factor});
+    AsmWriter{0x004309DE}.fmul<float>(AsmRegMem{&scanner_sensitivity_factor});
 }
 
 void update_scope_sensitivity()
 {
-    // prevent division by zero if somehow the modifier gets set to 0
-    float modifier = std::clamp(static_cast<float>(g_game_config.scope_sensitivity_modifier), 0.01f, 4.0f);
+    float modifier = clamp_sensitivity_modifier(g_game_config.scope_sensitivity_modifier);
     scope_sensitivity_factor = rf::scope_sensitivity_constant * (1.0f / modifier);
 }
 
 void update_scanner_sensitivity()
 {
-    float modifier = std::clamp(static_cast<float>(g_game_config.scanner_sensitivity_modifier), 0.01f, 4.0f);
+    float modifier = clamp_sensitivity_modifier(g_game_config.scanner_sensitivity_modifier);
     scanner_sensitivity_factor = rf::scanner_sensitivity_constant * modifier;
 }
 
 ConsoleCommand2 scope_sens_cmd{
     "scope_sensitivity_modifier",
     [](std::optional<float> value_opt) {
         if (value_opt) {
-            float value = std::clamp(value_opt.value(), 0.01f, 4.0f);
+            float value = clamp_sensitivity_modifier(value_opt.value());
```

Comment:
We clamp here but it is not clamped upon loading from registry so either remove this clamping (prefer) or add clamping upon loading from registry.

Reply:
The big reason I had done it that way was to ensure that when a player enters just `scope_sensitivity_modifier`, the printed value was clamped. I'm realizing now though that I could just do the clamp when that is requested rather than every time the value is entered, which is more efficient and enables the removal of an unneeded local variable in each. Updated.